src/main/java/uk/gov/hmcts/ccd/domain/service/common/CaseAccessGroupUtils.java

```diff
@@ -0,0 +1,236 @@
+package uk.gov.hmcts.ccd.domain.service.common;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.hmcts.ccd.domain.model.definition.AccessTypeRoleDefinition;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseAccessGroup;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseAccessGroupWithId;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseDetails;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseTypeDefinition;
+import uk.gov.hmcts.ccd.endpoint.exceptions.ValidationException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.UUID;
+
+public class CaseAccessGroupUtils {
+
+    private static final Logger LOG = LoggerFactory.getLogger(CaseAccessGroupUtils.class);
+    public static final String CASE_ACCESS_GROUPS = "CaseAccessGroups";
+
+    protected static final String CCD_ALL_CASES = "CCD:all-cases-access";
+
+    protected static final String ORGANISATION = "Organisation";
+    protected static final String ORGANISATIONID = "OrganisationID";
+    protected static final String ORG_IDENTIFIER_TEMPLATE = "$ORGID$";
+    protected static final String ORG_POLICY_CASE_ASSIGNED_ROLE = "OrgPolicyCaseAssignedRole";
+    protected static final String CASE_ACCESS_GROUP_TYPE = "caseAccessGroupType";
+    protected static final String GROUPACCESS_VALUE = "value";
+    protected static final String GROUPACCESS_ID = "id";
+
+    private CaseDataService caseDataService;
+    private CaseTypeDefinition caseTypeDefinition;
```

Comment:
This class should really be a bean (e.g. `@Component`), inject `CaseDataService` (since it's already a service, we shouldn't pass through from the invoking class, and remove `CaseTypeDefinition` from the class scope.

Reply:
fixed in #2389